src/representer/mapping.nim

```diff
@@ -0,0 +1,45 @@
+import hashes, tables, strutils
+
+type
+  NormalizedIdent = string
+  IdentMap* = OrderedTable[NormalizedIdent, string]
+
+proc hash(x: NormalizedIdent): Hash {.used.} =
+  !$(x[0].hash !& x[1..^1].hashIgnoreStyle)
+
+proc `==`(a, b: NormalizedIdent): bool =
+  a[0] == b[0] and
+    a.replace("_", "").toLowerAscii == b.replace("_", "").toLowerAscii
+
+
+when isMainModule:
+  template setup: untyped {.dirty.} =
+    var map: IdentMap
+
+  block testInitialization:
+    setup
+    doAssert map.len == 0
+
+  block insertIdentifier:
+    setup
+    map["x"] = "placeholder_1"
+    map["y"] = "placeholder_2"
+    doAssert map.len == 2
+  
+  block firstLetterCapital:
+    setup
+    map["x"] = "placeholder_0"
+    map["X"] = "placeholder_1"
+    echo map
+    echo map.len
+    doAssert map.len == 2
+
+  block otherLetterCapitals:
+    setup
+    map["hello"] = "placeholder_1"
+    map["hElLO"] = "placeholder_100"
+    map["Hello"] = "placeholder_2"
+    map["HELLo"] = "placeholder_200"
+    echo map
+    echo map.len
+    doAssert map.len == 2
```

Comment:
Currently this raises an assertion defect because `map.len` is 4.

Reply:
Should be fixed with  [a1d2717](https://github.com/exercism/nim-representer/pull/3/commits/a1d271725e855cfa065afe28d556892a019a7cdc)